nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5048,6 +5058,201 @@ async fn updates_refresh(
+        Ok(HttpResponseOk(SystemVersion {
+            version_range: VersionRange {
+                low: SemverVersion::new(0, 0, 1),
+                high: SemverVersion::new(0, 0, 2),
+            },
+            status: VersionStatus::Steady {
+                reason: VersionSteadyReason::Completed,
+            },
+        }))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// View version and update status of component tree
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/components",
+     tags = ["system"],
+}]
+async fn system_component_version_list(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<UpdateableComponent>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let query = query_params.into_inner();
+    let pagparams = data_page_params_for(&rqctx, &query)?;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let components = nexus
+            .updateable_components_list_by_id(&opctx, &pagparams)
+            .await?
+            .into_iter()
+            .map(|u| u.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            components,
+            &|_, u: &UpdateableComponent| u.identity.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// List all updates
+#[endpoint {
+     method = GET,
+     path = "/v1/system/update/updates",
+     tags = ["system"],
+}]
+async fn system_update_list(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    // TODO: pagination should probably be by most recent first, ID is nonsense
+    query_params: Query<PaginatedById>,
+) -> Result<HttpResponseOk<ResultsPage<SystemUpdate>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let query = query_params.into_inner();
+    let pagparams = data_page_params_for(&rqctx, &query)?;
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+        let updates = nexus
+            .system_updates_list_by_id(&opctx, &pagparams)
+            .await?
+            .into_iter()
+            .map(|u| u.into())
+            .collect();
+        Ok(HttpResponseOk(ScanById::results_page(
+            &query,
+            updates,
+            &|_, u: &SystemUpdate| u.identity.id,
+        )?))
+    };
+    apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
+}
+
+/// Path parameters for SystemUpdate requests
+#[derive(Deserialize, JsonSchema)]
+struct SystemUpdatePathParam {
+    // TODO: do updates have names? Should this therefore be NameOrId? Do we
+    // allow access by some other string identifier (like version) which, like
+    // Name, is disjoint with the set of Uuids? NameOrVersion?
+    id: Uuid,
```

Comment:
I definitely think we _wouldn't_ do `NameOrId` here. It doesn't, in my mind, make sense for an update to have a name. `IdOrVersion` could be a thing, but there's actually quite a bit of boilerplate around making something like that work so we should just proceed with caution there.